sdk/core/core/src/az_http_policy.c

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <az_http_policy.h>
+
+#include <az_contract.h>
+#include <az_result.h>
+#include <az_str.h>
+
+#include <az_http_pipeline.h>
+
+#include <_az_cfg.h>
+
+AZ_NODISCARD AZ_INLINE az_result az_http_pipeline_nextpolicy(
+    az_http_policy * const p_policies,
+    az_http_request_builder * const hrb,
+    az_mut_span const * const response) {
+
+  AZ_CONTRACT_ARG_NOT_NULL(p_policies);
+  AZ_CONTRACT_ARG_NOT_NULL(hrb);
+  AZ_CONTRACT_ARG_NOT_NULL(response);
+
+  //Transport Policy is the last policy in the pipeline
+  //  it returns without calling nextpolicy
+  if (&(p_policies[0]) == NULL || p_policies[0].pfnc_process == NULL) {
+    return AZ_ERROR_HTTP_PIPELINE_INVALID_POLICY;
+  }
+
+  return p_policies[0].pfnc_process(&(p_policies[1]), hrb, response);
+}
+
+static az_span const AZ_MS_CLIENT_REQUESTID = AZ_CONST_STR("x-ms-client-request-id");
+
+AZ_NODISCARD az_result az_http_pipeline_policy_uniquerequestid(
+    az_http_policy * const p_policies,
+    az_http_request_builder * const hrb,
+    az_mut_span const * const response) {
+  AZ_CONTRACT_ARG_NOT_NULL(p_policies);
+  AZ_CONTRACT_ARG_NOT_NULL(hrb);
+  AZ_CONTRACT_ARG_NOT_NULL(response);
+
+  //TODO - add a UUID create implementation
+  az_span const uniqueid = AZ_CONST_STR("123e4567-e89b-12d3-a456-426655440000");
+
+
+  // Append the Unique GUID into the headers
+  //  x-ms-client-request-id
+  az_result add_header_result = az_http_request_builder_append_header(
```

Comment:
AZ_RETURN_IF_FAILED, or at least `az_result const add_header_result`

Reply:
I've been using patterns like:
  if (az_failed(add_header_result)) {
    return add_header_result;
  }

I can switch to the macro since it does the same